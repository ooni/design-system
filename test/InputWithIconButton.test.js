/* global describe, test, expect, jest */
import React from 'react'
import { fireEvent } from '@testing-library/react'

import { renderWithTheme } from './index'
import { InputWithIconButton } from '../'

describe('InputWithIconButton', () => {
  test('renders', () => {
    const { container } = renderWithTheme(
      <InputWithIconButton icon={<div />} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('render with onAction prop', () => {
    const { container } = renderWithTheme(
      <InputWithIconButton onAction={() => {}} icon={<div />} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('onAction is called on click', () => {
    const onAction = jest.fn()
    const { container, getByRole } = renderWithTheme(
      <InputWithIconButton onAction={onAction} data-testid='test-input' />
    )
    expect(container.firstChild).toMatchSnapshot()

    const inputWithButton = getByRole('button')
    fireEvent.click(inputWithButton)
    expect(onAction).toHaveBeenCalled()
  })
})
