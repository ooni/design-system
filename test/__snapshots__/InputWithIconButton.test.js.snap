// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputWithIconButton onAction is called on click 1`] = `
.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 90%;
  margin-top: 8px;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: default;
  color: inherit;
  background-color: transparent;
  font-family: inherit;
  display: block;
  vertical-align: middle;
  background-color: transparent;
  border: none;
  border-radius: 0;
  border-bottom-style: solid;
  border-bottom-width: 1px;
  outline: none;
  margin: 0;
  padding: 0;
  width: 100%;
  padding-top: 8px;
  padding-bottom: 8px;
  box-shadow: none;
  box-sizing: content-box;
  -webkit-transition: all .3s;
  transition: all .3s;
  border-bottom-color: #f1f3f5;
  font-size: 16px;
}

.c2:hover {
  border-bottom-color: #f1f3f5;
}

.c2:focus {
  border-bottom-color: #0588cb;
}

.c3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 10%;
}

.c4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: #f8f9fa;
  background-color: #0588cb;
  border: 0;
  border-radius: 4px;
  display: inline-block;
  padding-left: 16px;
  padding-right: 16px;
  text-transform: capitalize;
  text-align: center;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  -webkit-transition: .2s ease-out;
  transition: .2s ease-out;
  outline: 0;
  color: #adb5bd;
  background-color: transparent;
  border-radius: none;
  border-width: 2px;
  border-style: solid;
  border-color: #0588cb;
  border: none;
  border-width: 2px;
  padding-left: 16px;
  padding-right: 16px;
  font-size: 16px;
}

.c4:hover:enabled {
  border-color: #37a6ed;
}

.c4:active:enabled {
  border-color: #0f77b8;
}

.c4:enabled {
  cursor: pointer;
}

.c4:hover {
  color: #ced4da;
}

.c4:active {
  color: #868e96;
}

.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-align-items: top;
  -webkit-box-align: top;
  -ms-flex-align: top;
  align-items: top;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  class="c0"
>
  <div
    class="c1"
    width="0.9"
  >
    <div>
      <input
        class="c2"
        data-testid="test-input"
        font-size="1"
        rows="3"
        type="text"
      />
    </div>
  </div>
  <div
    class="c3"
    width="0.1"
  >
    <button
      class="c4"
      font-size="1"
    />
  </div>
</div>
`;

exports[`InputWithIconButton render with onAction prop 1`] = `
.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 90%;
  margin-top: 8px;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: default;
  color: inherit;
  background-color: transparent;
  font-family: inherit;
  display: block;
  vertical-align: middle;
  background-color: transparent;
  border: none;
  border-radius: 0;
  border-bottom-style: solid;
  border-bottom-width: 1px;
  outline: none;
  margin: 0;
  padding: 0;
  width: 100%;
  padding-top: 8px;
  padding-bottom: 8px;
  box-shadow: none;
  box-sizing: content-box;
  -webkit-transition: all .3s;
  transition: all .3s;
  border-bottom-color: #f1f3f5;
  font-size: 16px;
}

.c2:hover {
  border-bottom-color: #f1f3f5;
}

.c2:focus {
  border-bottom-color: #0588cb;
}

.c3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 10%;
}

.c4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: #f8f9fa;
  background-color: #0588cb;
  border: 0;
  border-radius: 4px;
  display: inline-block;
  padding-left: 16px;
  padding-right: 16px;
  text-transform: capitalize;
  text-align: center;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  -webkit-transition: .2s ease-out;
  transition: .2s ease-out;
  outline: 0;
  color: #adb5bd;
  background-color: transparent;
  border-radius: none;
  border-width: 2px;
  border-style: solid;
  border-color: #0588cb;
  border: none;
  border-width: 2px;
  padding-left: 16px;
  padding-right: 16px;
  font-size: 16px;
}

.c4:hover:enabled {
  border-color: #37a6ed;
}

.c4:active:enabled {
  border-color: #0f77b8;
}

.c4:enabled {
  cursor: pointer;
}

.c4:hover {
  color: #ced4da;
}

.c4:active {
  color: #868e96;
}

.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-align-items: top;
  -webkit-box-align: top;
  -ms-flex-align: top;
  align-items: top;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  class="c0"
>
  <div
    class="c1"
    width="0.9"
  >
    <div>
      <input
        class="c2"
        font-size="1"
        rows="3"
        type="text"
      />
    </div>
  </div>
  <div
    class="c3"
    width="0.1"
  >
    <button
      class="c4"
      font-size="1"
    >
      <div />
    </button>
  </div>
</div>
`;

exports[`InputWithIconButton renders 1`] = `
.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 90%;
  margin-top: 8px;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: default;
  color: inherit;
  background-color: transparent;
  font-family: inherit;
  display: block;
  vertical-align: middle;
  background-color: transparent;
  border: none;
  border-radius: 0;
  border-bottom-style: solid;
  border-bottom-width: 1px;
  outline: none;
  margin: 0;
  padding: 0;
  width: 100%;
  padding-top: 8px;
  padding-bottom: 8px;
  box-shadow: none;
  box-sizing: content-box;
  -webkit-transition: all .3s;
  transition: all .3s;
  border-bottom-color: #f1f3f5;
  font-size: 16px;
}

.c2:hover {
  border-bottom-color: #f1f3f5;
}

.c2:focus {
  border-bottom-color: #0588cb;
}

.c3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 10%;
}

.c4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: #f8f9fa;
  background-color: #0588cb;
  border: 0;
  border-radius: 4px;
  display: inline-block;
  padding-left: 16px;
  padding-right: 16px;
  text-transform: capitalize;
  text-align: center;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  -webkit-transition: .2s ease-out;
  transition: .2s ease-out;
  outline: 0;
  color: #adb5bd;
  background-color: transparent;
  border-radius: none;
  border-width: 2px;
  border-style: solid;
  border-color: #0588cb;
  border: none;
  border-width: 2px;
  padding-left: 16px;
  padding-right: 16px;
  font-size: 16px;
}

.c4:hover:enabled {
  border-color: #37a6ed;
}

.c4:active:enabled {
  border-color: #0f77b8;
}

.c4:enabled {
  cursor: pointer;
}

.c4:hover {
  color: #ced4da;
}

.c4:active {
  color: #868e96;
}

.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-align-items: top;
  -webkit-box-align: top;
  -ms-flex-align: top;
  align-items: top;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  class="c0"
>
  <div
    class="c1"
    width="0.9"
  >
    <div>
      <input
        class="c2"
        font-size="1"
        rows="3"
        type="text"
      />
    </div>
  </div>
  <div
    class="c3"
    width="0.1"
  >
    <button
      class="c4"
      font-size="1"
    >
      <div />
    </button>
  </div>
</div>
`;
